### PR TITLE
UpdateJob when Job annotations or labels changed

### PR DESCRIPTION
**Pull request by silenceli**
bug description: https://github.com/volcano-sh/volcano/issues/805

### CODE REVIEW

File: pkg/controllers/job/job_controller_handler.go [hunk 1]
```diff
@@ -85,8 +85,13 @@ func (cc *Controller) updateJob(oldObj, newObj interface{}) {
 
 	// NOTE: Since we only reconcile job based on Spec, we will ignore other attributes
 	// For Job status, it's used internally and always been updated via our controller.
-	if reflect.DeepEqual(newJob.Spec, oldJob.Spec) && newJob.Status.State.Phase == oldJob.Status.State.Phase {
-		klog.V(6).Infof("Job update event is ignored since no update in 'Spec'.")
+	// NOTE: Maybe we should consider annotations and labels
+	if reflect.DeepEqual(newJob.Annotations, oldJob.Annotations) &&
+		reflect.DeepEqual(newJob.Labels, oldJob.Labels) &&
+		reflect.DeepEqual(newJob.Spec, oldJob.Spec) &&
+		newJob.Status.State.Phase == oldJob.Status.State.Phase {
```
[the remark this answers is not in the document]
Comment (silenceli) [reply pkg/controllers/job/job_controller_handler.go]: Thx，all done

**Comment by silenceli** [comment]
Please review again, thanks~

**Comment by hzxuzhonghu** [comment]
@Thor-wl please help cherry-pick to release 0.4 branch

**Comment by Thor-wl** [comment]
> @Thor-wl please help cherry-pick to release 0.4 branch

done. PR https://github.com/volcano-sh/volcano/pull/806#issuecomment-654133510

**Pull request merged**